app/src/main/java/com/babylon/wallet/android/data/repository/cache/database/StateDao.kt

```diff
@@ -204,6 +204,17 @@ interface StateDao {
     )
     fun getOwnedNfts(accountAddress: AccountAddress, resourceAddress: ResourceAddress, stateVersion: Long): List<NFTEntity>
 
+    @Query(
+        """
+        SELECT NFTEntity.* FROM AccountNFTJoin
+        INNER JOIN NFTEntity ON AccountNFTJoin.resource_address = NFTEntity.address AND AccountNFTJoin.local_id = NFTEntity.local_id
+        WHERE 
+            AccountNFTJoin.account_address = :accountAddress AND 
+            AccountNFTJoin.resource_address = :resourceAddress 
+        """
+    )
+    fun getOwnedNfts(accountAddress: AccountAddress, resourceAddress: ResourceAddress): List<NFTEntity>
```

Comment:
This takes from the db the latest NFT items (regardless of the state version) in order to be able to filter out the hidden items or the collection altogether if all of its items (owned by the user) are hidden.

Reply:
I'm not sure about this solution, please point out anything that might go wrong.